### PR TITLE
feat: Add NVIDIA triton trt-llm extension

### PR DESCRIPTION
**Pull request by vuonghoainam**
For #821 

**Integration diagram**
Image

**NVIDIA triton inference server and TensorRT LLM setup**
- https://github.com/hamelsmu/llama-inference/blob/master/triton-tensorRT/README.md
- This can be extended to `triton-inference-cluster` using Helm on Kubernetes on DGX clusters

### CODE REVIEW

**Comment by vuonghoainam** [comment]
1. Check if extension installed - See `Inference Triton Trt Llm Extension - v 1.0.0`
CleanShot 2023-12-07 at 09 07 41

2. Find model in Hub
CleanShot 2023-12-07 at 09 07 53

3. Update `~/jan/engines/triton_trtllm.json` with `base_url` as remote public/ private IP
CleanShot 2023-12-07 at 09 16 08

5. Chat with the remote `llama2 7b` model on remote `NVIDIA Triton inference server`
CleanShot 2023-12-07 at 09 09 45

**Comment by vuonghoainam** [comment]
There is a current blockage is the error on Triton inference server + trt llm with missing space character: https://github.com/triton-inference-server/tensorrtllm_backend/issues/34
I will check the answer to follow up.

**Comment by 0xSage** [comment]
What's the rationale for having both `inference-extension` (which just has the nitro binary) and `inference-nitro-extension`?

**Comment by vuonghoainam** [comment]
No, the `inference-extension` has been removed @0xSage

**Review by tikikun — APPROVED** [review]
LGTM

File: extensions/inference-triton-trtllm-extension/src/@types/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Model } from "@janhq/core";
+
+declare const MODULE: string;
```
Comment (louis-jan) [inline extensions/inference-triton-trtllm-extension/src/@types/global.d.ts:3]: remove MODULE

File: extensions/inference-triton-trtllm-extension/src/helpers/sse.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { Observable } from "rxjs";
```
Comment (louis-jan) [inline extensions/inference-triton-trtllm-extension/src/helpers/sse.ts:1]: Have you tried running this? I don't see rsjs defined

File: extensions/inference-triton-trtllm-extension/src/helpers/sse.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      temperature: 0,
+      bad_words: "",
+      stop_words: "[DONE]",
+      stream: true
+    });
+    fetch(`${engine.base_url}/v2/models/ensemble/generate_stream`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "text/event-stream",
```
Comment (louis-jan) [inline extensions/inference-triton-trtllm-extension/src/helpers/sse.ts:32]: Does it support non-streaming request? We just added this support onto latest main

**Review by louis-jan — APPROVED** [review]

File: extensions/inference-triton-trtllm-extension/src/@types/global.d.ts (hunk 1 above)
Comment (vuonghoainam) [reply extensions/inference-triton-trtllm-extension/src/@types/global.d.ts:3]: Done, thanks

File: extensions/inference-triton-trtllm-extension/src/helpers/sse.ts (hunk 3 above)
Comment (vuonghoainam) [reply extensions/inference-triton-trtllm-extension/src/helpers/sse.ts:32]: Yes, it supports both streaming/ non-streaming response

**Pull request merged**